src/main/scala/fi/oph/koski/raportit/LukioRaporttiOrdering.scala

```diff
@@ -0,0 +1,81 @@
+package fi.oph.koski.raportit
+
+import fi.oph.koski.suoritusote.KoulutusModuuliOrdering.järjestäSuffiksinMukaan
+
+object LukioRaporttiOppiaineetOrdering extends Ordering[LukioRaporttiOppiaineJaKurssit] {
+  override def compare(x: LukioRaporttiOppiaineJaKurssit, y: LukioRaporttiOppiaineJaKurssit): Int = {
+    orderByPaikallisuus(x.oppiaine, y.oppiaine)
+  }
+
+  private def orderByPaikallisuus(x: LukioRaporttiOppiaine, y: LukioRaporttiOppiaine) = (x.koulutusmoduuliPaikallinen, y.koulutusmoduuliPaikallinen) match {
+    case (true, true) => orderByNimi(x, y)
+    case (false, false) => orderByKoulutusmoduulikoodiarvo(x, y)
+    case (a, b) => a compare b
+  }
+
+  private def orderByKoulutusmoduulikoodiarvo(x: LukioRaporttiOppiaine, y: LukioRaporttiOppiaine) = {
+    koodiarvonJärjestysnumero(x) compare koodiarvonJärjestysnumero(y) match {
+      case 0 => orderByNimi(x, y)
+      case order => order
+    }
+  }
+
+  private def orderByNimi(x: LukioRaporttiOppiaine, y: LukioRaporttiOppiaine) = {
+    if (x.koulutusmoduuliKoodiarvo == "MA") {
+      orderMatikkaByOppimäärä(x, y)
+    } else {
+      x.nimi.capitalize compare y.nimi.capitalize
+    }
+  }
+
+  private def orderMatikkaByOppimäärä(x: LukioRaporttiOppiaine, y: LukioRaporttiOppiaine) = {
```

Comment:
Toimiiko tämä nyt jotenkin väärin? Ordering:in pitäisi toimia niin, että kaksi samaa arvoa palauttaa 0:n.
```
    val foo = LukioRaporttiOppiaineJaKurssit(LukioRaporttiOppiaine("Matematiikka", "foo", koulutusmoduuliPaikallinen = true), Nil)
    LukioRaporttiOppiaineetOrdering.compare(foo, foo) // => 0 as expected
    val math = LukioRaporttiOppiaineJaKurssit(LukioRaporttiOppiaine("Pitkä matematiikka", "MA", koulutusmoduuliPaikallinen = true), Nil)
    LukioRaporttiOppiaineetOrdering.compare(math, math) // => 1, expected 0
    val math2 = LukioRaporttiOppiaineJaKurssit(LukioRaporttiOppiaine("pitkä matematiikka", "MA", koulutusmoduuliPaikallinen = true), Nil)
    LukioRaporttiOppiaineetOrdering.compare(math2, math2) // => -1, expected 0, why changing to lowercase changes the result?
```